lib/parser/index.js

```diff
@@ -0,0 +1,79 @@
+const sketch2json = require('sketch2json');
+const { getClient, figma2jsonFromAPI, findIds } = require('../figma');
+const {
+  getPageArrays,
+  getColorsFromArtboard,
+  getGradientsFromArtboard,
+  getShadowsFromArtboard,
+  getBordersFromArtboard,
+} = require('../../lib/sketch');
+const { readFile, uniqueArray } = require('../utils');
+
+const parseTokens = async (args, flags) => {
+  const [file] = args;
+  const isSketch = /.*sketch/.test(file);
+  if (isSketch) {
+    try {
+      const response = await sketch2json(await readFile(file));
+      const primitivesPage = getPageArrays(response).find(
+        i => i.name.toLowerCase() === 'primitives',
+      );
+      if (!primitivesPage) {
+        throw new Error(`No primitives page found.`);
+      }
+
+      return {
+        colors: flags.useColorArtboards
+          ? getColorsFromArtboard(primitivesPage.layers)
+          : response.document.assets.colorAssets.map(({ color, _class, name }) => ({
+              ...color,
+              _class,
+              name,
+            })),
+        gradients: flags.useGradientArtboards
+          ? getGradientsFromArtboard(primitivesPage.layers)
+          : response.document.assets.gradientAssets.map(({ gradient, _class }) => ({
+              ...gradient,
+              _class,
+            })),
+        textStyles: response.document.layerTextStyles,
+        shadows: getShadowsFromArtboard(primitivesPage.layers),
+        borders: getBordersFromArtboard(primitivesPage.layers),
+        fonts: response.meta.fonts,
+        sketchVersion: response.meta.appVersion,
+        response,
+      };
+    } catch (err) {
+      throw new Error(err);
+    }
+  } else {
+    const { token } = flags;
+    if (!token) {
+      throw new Error('Please add a Figma API authorization token');
+    }
+
+    const client = getClient(token);
+    const response = await figma2jsonFromAPI(client, file);
+
+    try {
```

Comment:
Why would this need to be wrapped in a trycatch?

Seems like a simple return of an object using functions which only derive data and do not perform any async logic/